Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/contests-filters/ContestFilters.tsx

```diff
@@ -35,16 +41,34 @@ const ContestFilters = ({ onFilterClick }: IContestFiltersProps) => {
     const { state: { possibleFilters } } = useContests();
 
     const handleFilterClick = useCallback(
-        (filterId: number) => {
+        (filterId: number, filterType: FilterType) => {
             const filter = possibleFilters.find(({ id }) => filterId === id);
 
             if (isNil(filter)) {
                 return;
             }
 
             onFilterClick(filter);
+
+            if (FilterType.Status === filterType) {
+                if (previousSelectedStatusButton === filterId) {
+                    setSelectedStatusFilter(defaultState.id);
+                    setPreviousSelectedStatusButton(defaultState.id);
+                } else {
+                    setSelectedStatusFilter(filterId);
+                    setPreviousSelectedStatusButton(filterId);
+                }
+            } else if (FilterType.Strategy === filterType) {
+                if (previousSelectedStrategyButton === filterId) {
+                    setSelectedStrategyFilter(0);
+                    setPreviousSelectedStrategyButton(0);
+                } else {
+                    setSelectedStrategyFilter(filterId);
+                    setPreviousSelectedStrategyButton(filterId);
+                }
+            }
```

Comment:
Simplify this. As a rule, nested `if-else` is prohibited, unless absolutely necessary.